Digipost.Api.Client.Shared/Certificate/CertificateChainValidator.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Digipost.Api.Client.Shared.Extensions;
+using Digipost.Api.Client.Shared.Resources.Language;
+using static Digipost.Api.Client.Shared.Resources.Language.LanguageResource;
+
+namespace Digipost.Api.Client.Shared.Certificate
+{
+    public class CertificateChainValidator
+    {
+        public CertificateChainValidator(X509Certificate2Collection certificateStore)
+        {
+            CertificateStore = certificateStore;
+        }
+
+        public X509Certificate2Collection CertificateStore { get; set; }
+
+        [Obsolete("Use CertificateStore instead.")]
+        public X509Certificate2Collection SertifikatLager => CertificateStore;
+
+        /// <summary>
+        ///     Validerer sertifikatkjeden til sertifikatet. Gjør dette ved å validere mot <see cref="SertifikatLager" />
+        /// </summary>
+        /// <param name="certificate"></param>
+        /// <returns></returns>
+        [Obsolete("Use IsValidChain instead.")]
+        public bool ErGyldigSertifikatkjede(X509Certificate2 certificate)
```

Comment:
Er litt norsk her også, men det ser ut til å være meningen for å ivareta bakoverkompatibiliteten, eller misforstår jeg?